src/test/ui/parser/bare-struct-body.stderr

```diff
@@ -14,6 +14,23 @@ LL |     val: (),
 LL | } }
    |
 
+error: this code is interpreted as a block expression, not an array
+  --> $DIR/bare-struct-body.rs:12:13
+   |
+LL |       let x = {
+   |  _____________^
+LL | |         val: (),
+LL | |     };
+   | |_____^
+   |
+   = note: to define an array, one would use square brackets instead of curly braces
+help: try using [] instead of {}
+   |
+LL |     let x = [
+LL |         val: (),
+LL |     ];
+   |
```

Comment:
We probably wouldn't want this error or the one above, as we are emitting two diagnostics for the same syntax error. `maybe_suggest_brackets_instead_of_braces` doesn't signal whether this worked or not, so instead what it could do is return a `PResult` and you can call `self.maybe_suggest_brackets_instead_of_braces(lo)?` instead.

Reply:
By actually parsing as if they were arrays, this is no longer emitted.